relay-config/src/config.rs

```diff
@@ -797,8 +799,14 @@ pub struct TopicAssignments {
     pub outcomes_billing: Option<TopicAssignment>,
     /// Session health topic name.
     pub sessions: TopicAssignment,
-    /// Metrics topic name.
+    /// Default topic name for all aggregate metrics. Specialized topics for session-based and
+    /// transaction-based metrics can be configured via `metrics_sessions` and
+    /// `metrics_transactions` each.
     pub metrics: TopicAssignment,
+    /// Topic name for metrics extracted from sessions. Defaults to the assignment of `metrics`.
+    pub metrics_sessions: Option<TopicAssignment>,
+    /// Topic name for metrics extracted from transactions. Defaults to the assignment of `metrics`.
+    pub metrics_transactions: Option<TopicAssignment>,
```

Comment:
_nit_: If we introduce a MetricNamespace enum, this could become a mapping of `MetricNamespace` => `TopicAssignment`, which would allow for more generic code in all other places. Provided that all unknown namespace strings are mapped to `MetricNamespace::Unknown`, this would even allow to create config before updating Relay. 

I have not fully thought through all implications here, however.